bot.ml

```diff
@@ -786,15 +786,145 @@ let callback _conn req body =
+                            ~message:
+                              (f
+                                 "@%s: PR cannot be merged as it isn't \
+                                  approved yet."
+                                 comment_info.author)
+                            ~id:pr.id
+                      | APPROVED -> (
+                          if
+                            not
+                              (List.exists comment_info.issue.assignees
+                                 ~f:(fun login ->
+                                   String.equal login comment_info.author))
+                          then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR as you're not \
+                                    among the assignees."
+                                   comment_info.author)
+                              ~id:pr.id
+                          else if String.equal comment_info.author pr.user then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: You can't merge the PR since you \
+                                    authored it."
+                                   comment_info.author)
+                              ~id:pr.id
+                          else if
+                            not (String.equal reviews_info.baseRef "master")
+                          then
+                            GitHub_mutations.post_comment ~bot_info
+                              ~message:
+                                (f
+                                   "@%s: PR targets branch %s instead of \
+                                    master, abort merge."
+                                   comment_info.author reviews_info.baseRef)
+                              ~id:pr.id
+                          else
+                            ( match
+                                List.fold_left ~init:[] reviews_info.files
+                                  ~f:(fun acc f ->
+                                    if
+                                      string_match
+                                        ~regexp:"dev/ci/user-overlays/\\(.*\\)"
+                                        f
+                                    then Str.matched_group 1 f :: acc
+                                    else acc)
+                              with
+                            | [] ->
+                                Lwt.return ()
+                            | overlays ->
+                                GitHub_mutations.post_comment ~bot_info
+                                  ~message:
+                                    (f
+                                       "@%s: Please take of the following \
```

Comment:
```suggestion
                                       "@%s: Please take care of the following \
```

Furthermore, I would delay this step after the PR merging, since it isn't blocking it.